### DISCUSSION TITLE
Iterations close too early; no current iteration for a whole day

### DISCUSSION THREAD

**Discussion by bricker** [community/community#20032] [Projects and Issues] [answered]
## Description

The Iteration system closes iterations at the beginning of the end date, leaving a whole day with no "Current" iteration.

## Setup
- Auto-Create a new iteration, Iteration 1, using the "Add Iteration" button. For this example, end date is automatically set to July 5.
- Auto-Create another iteration, Iteration 2, also using the "Add Iteration" button. Start date is automatically set to the day after the previous iteration's end date, in this example July 6th.
- An iteration cannot start on or before the previous iteration's end date, whether created manually or automatically.

## Issue
The iteration is "Completed" some time at the _beginning_ of the end date. In this example, on the morning of July 5th (PT), Iteration A is marked as "Completed", but Iteration B doesn't start until July 6th. So there is nothing associated with `@current` for all of July 5th.

A few workarounds I tried that didn't work:
1. Manually moving Iteration B's start date to July 5th. Because an iteration can't start on or before the end date of the previous iteration, the UI doesn't let you do this.
2. Manually moving Iteration A's end date to July 6th. The UI lets you do this, but then automatically changes Iteration B's start date to July 7th, resulting in the same issue (but a day later).

**Comment by NevilleS** [comment]
This is the same as https://github.com/github-community/community/discussions/18834, but I like your description much better!

**Comment by 2percentsilk** [comment]
As I mentioned on other reports of this issue, I'm aware and investigating with the team! Will share when I know more.

**Comment by Neudrino** [comment]
There are several issues describing the same problem.

The **most up-voted** one is:
- https://github.com/github-community/community/discussions/20101

What was thought to be a **fix** was mentioned to be out for testing in:
- https://github.com/github-community/community/discussions/20101#discussioncomment-3149906

The others are:
- https://github.com/github-community/community/discussions/20217
- https://github.com/github-community/community/discussions/20032
- https://github.com/github-community/community/discussions/20108
- https://github.com/github-community/community/discussions/18834

**Comment by 2percentsilk** [comment, marked as answer]
Based on testing from the community and team, we believe we have fixed this bug. Thanks to everyone for reporting it!
